Add documentation for using private image registries and a default workspace image for Self hosted (#2640)
gitpod-io/website: gitpod/docs/self-hosted/latest/advanced/private-registries.md

---
section: self-hosted/latest
subsection: installation-guides
title: Private Registries
---

<script context="module">
  export const prerender = true;
</script>

# Using private image registries with Gitpod Self-Hosted

> This guide outlines the considerations to take when using images for workspaces that originate from private image registries. Please refer to the [docker image documentation](../../../config-docker) for more information about workspace images.

Reviewer: ```suggestion
This guide outlines the considerations to take when using images for workspaces that originate from private image registries. Please refer to the [Docker image documentation](../../../config-docker) for more information about workspace images.
```

> > Note: there are two ways of setting an image for a workspace: via the installation-wide [default `workspaceImage`](./default-workspace-image) and the [`.gitpod.yml` file](../../../config-gitpod-file). The latter overwrites the former.

Reviewer: ```suggestion
> Note: there are two ways of setting an image for a workspace: via the installation-wide [default `workspaceImage`](./default-workspace-image) and the [`.gitpod.yml` file](../../../config-gitpod-file). The `.gitpod.yml` file will take precedence if both are set.
```

As above

## Credentials

To access private registries, Gitpod needs to have the corresponding access credentials. These can be supplied in the KOTS installation UI:

![registry credentials UI](../../static/images/docs/self-hosted/private-registry-credentials.png)

## Allowlist

Gitpod also has an allow list for private base image registries. Any private image registry that is used throughout an installation needs to be on this allowlist.

> > **Important:** Any registry for which credentials are supplied is automatically added to this allowlist!

Reviewer 2: 💯 Stands out like we want.
